sirepo/template/controls.py

```diff
@@ -67,6 +67,23 @@ def write_parameters(data, run_dir, is_parallel):
     )
 
 
+def _add_monitor(data):
+    util = LatticeUtil(
+        data,
+        sirepo.sim_data.get_class('madx').schema(),
+    )
+    assert len(data.models.beamlines) == 1
+    b = data.models.beamlines[0]
+    if util.id_map[b['items'][-1]].type != 'MONITOR':
+        m = PKDict(
+            _id=LatticeUtil.max_id(data) + 1,
+            name='M1',
+            type='MONITOR',
+        )
+        data.models.elements.append(m)
+        b['items'].append(m._id)
+
```

Comment:
I simplified _add_monitor() a bit. The import process creates a single beamline, so all you need to do is check if the last item is a MONITOR.